pkg/cmd/util/clientcmd/factory.go

```diff
@@ -192,9 +192,14 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 				return nil, errors.New("provided options object is not a BuildLogOptions")
 			}
 			if bopts.Version != nil {
-				// should --version work with builds at all?
 				return nil, errors.New("cannot specify a version and a build")
 			}
+			if bopts.Previous {
+				version := buildutil.VersionForBuild(t.Name)
+				version--
```

Comment:
you should be checking if the version value that came back is <= 0 since you're using "0" to indicate problems parsing the version number, and of course a build version of "-1" isn't going to make sense.